### PR TITLE
feat: allow notification resource handler to get deployed within a vpc (closes #20903)

### LINKED ISSUE #20903: aws-s3: Cannot use notification resource handler in monitored environment
### Describe the bug

When the AWS does not allow for a lambda creation without an VPC attachment Bucket.addEventNotification() fails during the deployment of the Cloudformation Stack, because of the NotificationsResourceHandler lambda function.

### Expected Behavior

The props should contain a an optional parameter for a vpc assignment and the lambda should then get the correct vpc config attached.

### Current Behavior

The lambda gets deployed with no vpc attachment.

### Reproduction Steps

Restrict lambdas from being created without an VPC attachment in your account and then try to deploy:
`
const bucket = new Bucket(scope, id, {bucketName: "Test"})
const object_deleted_topic = new Topic(this, "object-deleted-topic", {
            displayName: "object-deleted",
        });
bucket.addEventNotification(EventType.OBJECT_REMOVED_DELETE, new SnsDestination(object_deleted_topic))
`

### Possible Solution

Add a new porperty to the bucket: notificationsHandlerVpc.

### Additional Information/Context

_No response_

### CDK CLI Version

2.29.1

### Framework Version

_No response_

### Node.js Version

16.15.1

### OS

Ubuntu 20.04

### Language

Typescript

### Language Version

3.9.10

### Other information

_No response_

### PR DESCRIPTION
**Pull request by falkvonohlen**
In an monitored AWS account it might be necessary to deploy any lambda inside a VPC. This causes the CDK deployment to fail, if Bucket.addEventNotification() is called, as it creates a utility lambda without a VPC attachment. This pull requests adds an optional attribute to the S3Bucket that can be used to provided a VPC to which the lambda gets deployed.

Fixes https://github.com/aws/aws-cdk/issues/20903
---

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
Before we can provide a meaningful review, we need the build to be passing.

Additionally, please make sure that your PR title confirms to the conventional commit standard (fix, feat, chore) and that it is written in a style that will reflect correctly in the change log (See [Contributing Guide, Pull Requests](https://github.com/aws/aws-cdk/blob/v1-main/CONTRIBUTING.md#step-4-pull-request))

**Comment by falkvonohlen** [comment]
Hi,

I don`t see why the build is failing.
It has something to do with the dependencies outside of the areas I touched.
Could someone please have a look on the build result and let me know what I need to!

Thanks Falk

**Comment by mrgrain** [comment]
@mergifyio update

**Comment by mrgrain** [comment]
@falkvonohlen Let's see if updating with the latest changes helps

File: packages/@aws-cdk/aws-amplify/package.json [hunk 1]
```diff
@@ -93,6 +93,7 @@
   "dependencies": {
     "@aws-cdk/aws-codebuild": "0.0.0",
     "@aws-cdk/aws-codecommit": "0.0.0",
+    "@aws-cdk/aws-ec2": "0.0.0",
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-amplify/package.json:96]: This is probably the source of the issue (and where you have it added below as well). Was adding this here a mistake? I don't see any other changes to `aws-amplify` so the dependency addition here doesn't make a lot of sense to me.

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
I noted inline where I _think_ the issue may be stemming from.

Additionally, please make sure that your PR title confirms to the conventional commit standard (fix, feat, chore) and that it is written in a style that will reflect correctly in the change log (See [Contributing Guide, Pull Requests](https://github.com/aws/aws-cdk/blob/v1-main/CONTRIBUTING.md#step-4-pull-request)). See other merged `feats` for examples of what we're looking for here.

**Comment by falkvonohlen** [comment]
I actually added the ec2 dependency in a separate commit as I thought it would solve my issue, but reverted it now.
However I still don`t see why the build is failing.

**Comment by TheRealAmazonKendra** [comment]
@Mergifyio update

**Comment by TheRealAmazonKendra** [comment]
This PR has been deemed to be abandoned, and will be closed. Please create a new PR for these changes if you think this decision has been made in error.

**Pull request closed without merging**